main.cpp

```diff
@@ -602,8 +602,9 @@ static bool add_endpoints(Mainloop &mainloop)
             return false;
         }
 
-        g_endpoints[i++] = udp.release();
-        mainloop.add_fd(udp->fd, g_endpoints[i], EPOLLIN);
+        g_endpoints[i] = udp.release();
```

Comment:
You do, because the udp object doesn't contain the pointer anymore. That was the reason why I used g_endpoints when calling add_fd().  I forgot about the `i` variable though, fixed it locally and didn't amend.